### PR TITLE
fix: Add a workflow to check the incoming push and PRs

### LINKED ISSUE #57: fix: adapting our build for the new version of proto-gen-go
Description:

There has been a version upgrade of the protobuf compiler and there has been certain changes which do not break the API in any way but makes some of the flags we used while compiling redundant. There is a need to change these in the makefile and accordingly update the readme.

Present Behaviour:

We are building using the older compiler

Expected Behaviour:

We should upgrade to the new compiler

### PR DESCRIPTION
**Pull request by littlestar642**
Description:

A github action was needed to check all the incoming PRs and pushes to master if they are able to successfully build as well all the tests pass. There is a need to upgrade the proto-gen-go compiler. The changes needed are mentioned in the README.md

fixes #57

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+
+  build:
+    name: Build
+    runs-on: macos-latest
```
Comment (faruqisan) [inline .github/workflows/go.yml:13]: Any reason why run on macOS? Is it going faster/slower or what?

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        go-version: ^1.13
+      id: go
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v2
+
+    - name: Get dependencies
+      run: |
+        go get -v -t -d ./...
+        if [ -f Gopkg.toml ]; then
```
Comment (faruqisan) [inline .github/workflows/go.yml:28]: we don't use dep anymore, remove this gopkg part.

File: .github/workflows/go.yml (hunk 1 above)
Comment (littlestar642) [reply .github/workflows/go.yml:13]: we have our machines as macOS on which we are presently building, therefore i chose to keep it macos. I guess if needed, we can for sure switch to ubuntu

File: .github/workflows/go.yml (hunk 2 above)
Comment (littlestar642) [reply .github/workflows/go.yml:28]: cool. That was suggested by actions to be included, so I did. I will remove it if not needed

**Review by yashipro13 — APPROVED** [review]

**Pull request merged**